jabgui/src/main/java/org/jabref/gui/walkthrough/WalkthroughOverlay.java

```diff
@@ -0,0 +1,147 @@
+package org.jabref.gui.walkthrough;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javafx.animation.KeyFrame;
+import javafx.animation.Timeline;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import javafx.stage.Window;
+import javafx.util.Duration;
+
+import org.jabref.gui.walkthrough.declarative.WindowResolver;
+import org.jabref.gui.walkthrough.declarative.step.WalkthroughStep;
+
+import org.jspecify.annotations.NonNull;
+import org.jspecify.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WalkthroughOverlay {
+    private static final Logger LOGGER = LoggerFactory.getLogger(WalkthroughOverlay.class);
+
+    private final Map<Window, SingleWindowWalkthroughOverlay> overlays = new HashMap<>();
+    private final Stage stage;
+    private final WalkthroughHighlighter walkthroughHighlighter;
+    private final Walkthrough walkthrough;
+    private @Nullable Timeline nodePollingTimeline;
```

Comment:
Im not yet convinced that we have to do polling and that this is not solvable by an event being fired by javafx. Polling takes unnecessary resources.